tensorflow/stream_executor/rocm/rocm_stream.cc

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/stream_executor/gpu/gpu_stream.h"
+
+#include "tensorflow/stream_executor/gpu/gpu_executor.h"
+#include "tensorflow/stream_executor/lib/status.h"
+#include "tensorflow/stream_executor/stream.h"
+
+namespace stream_executor {
+namespace gpu {
+
+bool GpuStream::Init() {
+  if (!GpuDriver::CreateStream(parent_->gpu_context(), &gpu_stream_)) {
+    return false;
+  }
+  return GpuDriver::CreateEvent(parent_->gpu_context(), &completed_event_,
+                                 GpuDriver::EventFlags::kDisableTiming)
+      .ok();
+}
+
+void GpuStream::Destroy() {
+  if (completed_event_ != nullptr) {
+    port::Status status =
+        GpuDriver::DestroyEvent(parent_->gpu_context(), &completed_event_);
+    if (!status.ok()) {
+      LOG(ERROR) << status.error_message();
+    }
+  }
+
+  GpuDriver::DestroyStream(parent_->gpu_context(), &gpu_stream_);
+}
+
+bool GpuStream::IsIdle() const {
+  return GpuDriver::IsStreamIdle(parent_->gpu_context(), gpu_stream_);
+}
+
+GpuStream *AsGpuStream(Stream *stream) {
```

Comment:
This file is trivially similar to cuda_stream as well.

Reply:
will merge `rocm/rocm_stream.cc` and `cuda/cuda_stream.cc` into a common `gpu/gpu_stream.cc` in the next commit